Source/Scene/Cesium3DTileBatchTable.js

```diff
@@ -0,0 +1,1519 @@
+            return true;
+        }
+
+        var offset = batchId * 2;
+        return (this._showAlphaProperties[offset] === 255);
+    };
+
+    var scratchColorBytes = new Array(4);
+
+    Cesium3DTileBatchTable.prototype.setColor = function(batchId, color) {
+        var featuresLength = this.featuresLength;
+        //>>includeStart('debug', pragmas.debug);
+        if (!defined(batchId) || (batchId < 0) || (batchId > featuresLength)) {
+            throw new DeveloperError('batchId is required and between zero and featuresLength - 1 (' + featuresLength - + ').');
+        }
+
+        if (!defined(color)) {
+            throw new DeveloperError('color is required.');
+        }
+        //>>includeEnd('debug');
+
+        if (Color.equals(color, Color.WHITE) && !defined(this._batchValues)) {
+            // Avoid allocating since the default is white
+            return;
+        }
+
+        var newColor = color.toBytes(scratchColorBytes);
+        var newAlpha = newColor[3];
+
+        var batchValues = getBatchValues(this);
+        var offset = batchId * 4;
+
+        var showAlphaProperties = getShowAlphaProperties(this);
+        var propertyOffset = batchId * 2;
+
+        if ((batchValues[offset] !== newColor[0]) ||
+            (batchValues[offset + 1] !== newColor[1]) ||
+            (batchValues[offset + 2] !== newColor[2]) ||
+            (showAlphaProperties[propertyOffset + 1] !== newAlpha)) {
+
+            batchValues[offset] = newColor[0];
+            batchValues[offset + 1] = newColor[1];
+            batchValues[offset + 2] = newColor[2];
+
+            var wasTranslucent = (showAlphaProperties[propertyOffset + 1] !== 255);
+
+            // Compute alpha used in the shader based on show and color.alpha properties
+            var show = showAlphaProperties[propertyOffset] !== 0;
+            batchValues[offset + 3] = show ? newAlpha : 0;
+            showAlphaProperties[propertyOffset + 1] = newAlpha;
+
+            // Track number of translucent features so we know if this tile needs
+            // opaque commands, translucent commands, or both for rendering.
+            var isTranslucent = (newAlpha !== 255);
+            if (isTranslucent && !wasTranslucent) {
+                ++this._translucentFeaturesLength;
+            } else if (!isTranslucent && wasTranslucent) {
+                --this._translucentFeaturesLength;
+            }
+
+            this._batchValuesDirty = true;
+        }
+    };
+
+    Cesium3DTileBatchTable.prototype.setAllColor = function(color) {
+        //>>includeStart('debug', pragmas.debug);
+        if (!defined(color)) {
+            throw new DeveloperError('color is required.');
+        }
+        //>>includeEnd('debug');
+
+        var featuresLength = this.featuresLength;
+        for (var i = 0; i < featuresLength; ++i) {
+            // PERFORMANCE_IDEA: duplicate part of setColor here to factor things out of the loop
```

Comment:
Yes they are not called often, just `debugColorizeTiles` and setting an `undefined` style. I removed the `PERFORMANCE_IDEA`s.